packages/monaco/lib/convert.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * @typedef {import('monaco-editor')} Monaco
+ * @typedef {import('monaco-editor').editor.IMarkerData} IMarkerData
+ * @typedef {import('monaco-editor').editor.IRelatedInformation} IRelatedInformation
+ * @typedef {import('monaco-editor').editor.ITextModel} ITextModel
+ * @typedef {import('monaco-editor').languages.typescript.Diagnostic} Diagnostic
+ * @typedef {import('monaco-editor').languages.typescript.DiagnosticRelatedInformation} DiagnosticRelatedInformation
+ * @typedef {import('monaco-editor').languages.CompletionItemKind} CompletionItemKind
+ * @typedef {import('monaco-editor').IRange} IRange
+ * @typedef {import('monaco-editor').MarkerSeverity} MarkerSeverity
+ * @typedef {import('monaco-editor').MarkerTag} MarkerTag
+ * @typedef {import('typescript').CompletionEntryDetails} CompletionEntryDetails
+ * @typedef {import('typescript').DiagnosticCategory} DiagnosticCategory
+ * @typedef {import('typescript').DiagnosticMessageChain} DiagnosticMessageChain
+ * @typedef {import('typescript').JSDocTagInfo} JSDocTagInfo
+ * @typedef {import('typescript').ScriptElementKind} ScriptElementKind
+ * @typedef {import('typescript').SymbolDisplayPart} SymbolDisplayPart
+ * @typedef {import('typescript').TextSpan} TextSpan
+ */
+
+/**
+ * Convert a TypeScript script element kind to a Monaco completion item kind.
+ *
+ * @param {Monaco} monaco
+ *   The Monaco editor module to use.
+ * @param {ScriptElementKind} kind
+ *   The TypeScript script element kind tp convert.
+ * @returns {CompletionItemKind}
+ *   The matching Monaco completion item kind.
+ */
+export function convertScriptElementKind(monaco, kind) {
```

Comment:
Wait, I think I read a lot of these functions before? Something to move to a shared place/utility?

Reply:
I think you mean `packages/language-server/lib/convert.js`. They are very similar, yet completely different. One converts to Monaco editor types, whereas the other converts to language server types. They have suble differences that make them incompatible.